pkg/api/vlabs/validate_test.go

```diff
@@ -829,6 +829,41 @@ func TestValidateImageNameAndGroup(t *testing.T) {
 	}
 }
 
+func TestMasterProfileValidate(t *testing.T) {
+	tests := []struct {
+		orchestratorType string
+		count            int
+		expectedErr      bool
+	}{
+		{
+			orchestratorType: OpenShift,
+			count:            1,
+		},
+		{
+			orchestratorType: OpenShift,
+			count:            3,
+			expectedErr:      true,
```

Comment:
tests need to test against a known, expected error in failure cases to really know you're hitting the code you think you are.  I'd also make the test provide the `MasterProfile` so this really can be used as a  `TestMasterProfileValidate` in the future and not just a count test.

Reply:
agreed